### PR TITLE
refs Task #148636 [common] init backend project

### PR DESCRIPTION
**Pull request by tashdidjonayed**
- init spring mvc project

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+	<name>Framgia Internal training management</name>
+	<packaging>war</packaging>
+	<version>1.0.0-BUILD-SNAPSHOT</version>
+	<properties>
+		<java-version>1.6</java-version>
+		<org.springframework-version>4.0.3.RELEASE</org.springframework-version>
+		<org.aspectj-version>1.7.4</org.aspectj-version>
+		<org.slf4j-version>1.7.5</org.slf4j-version>
+		<hibernate.version>4.3.5.Final</hibernate.version>
+	</properties>
```
Comment (VuThiTranVan) [inline pom.xml]: You should use the newer version: 
jdk 1.8
Spring 4.3.4.RELEASE
Hibernate 5.2.5.Final

File: src/main/java/com/framgia/fitm/dao/UserDAO.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+public interface UserDAO {
+	
+	public void login(int userId);
+	public void logout(int userId);
+	public void createUser(User user);
+	public void deleteUser(User user);
+	public void updateUser(User user);
+	public void showUser(User user);
+	public List<User> Index();
+
```
Comment (VuThiTranVan) [inline src/main/java/com/framgia/fitm/dao/UserDAO.java]: If these Class are not complete then ignore this comment.

1. method: createUser, deleteUser, updateUser, showUser with void ?

2. you should change name method [Index] -> such as: [listUser]

3. Hibernate#Session: it support method: SaveOrUpdate, delete, findById
You should using it and don't using createUser, deleteUser...

File: src/main/webapp/WEB-INF/spring/appServlet/servlet-context.xml [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+	</beans:bean>
+	<context:component-scan base-package="com.framgia.fitm.controller" />
+
+	<tx:annotation-driven transaction-manager="transactionManager" />
+
+	<beans:bean id="transactionManager"
+		class="org.springframework.orm.hibernate4.HibernateTransactionManager">
+		<beans:property name="sessionFactory"
+			ref="hibernate4AnnotatedSessionFactory" />
+	</beans:bean>
```
Comment (VuThiTranVan) [inline src/main/webapp/WEB-INF/spring/appServlet/servlet-context.xml]: Line 43 -> 83: create split into a config data sources file then import into
spring-datasource.xml

File: src/main/webapp/WEB-INF/spring/root-context.xml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<beans xmlns="http://www.springframework.org/schema/beans"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:security="http://www.springframework.org/schema/security"
+	xsi:schemaLocation="http://www.springframework.org/schema/security http://www.springframework.org/schema/security/spring-security-4.0.xsd
+		http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans.xsd">
+	
+	<!-- Root Context: defines shared resources visible to all other web components -->
+		
+</beans>
```
Comment (VuThiTranVan) [inline src/main/webapp/WEB-INF/spring/root-context.xml]: Is this file used to do? Is it necessary?

File: src/main/webapp/WEB-INF/views/person.jsp [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+<html>
+<head>
+	<title>Person Page</title>
+	<style type="text/css">
+		.tg  {border-collapse:collapse;border-spacing:0;border-color:#ccc;}
+		.tg td{font-family:Arial, sans-serif;font-size:14px;padding:10px 5px;border-style:solid;border-width:1px;overflow:hidden;word-break:normal;border-color:#ccc;color:#333;background-color:#fff;}
+		.tg th{font-family:Arial, sans-serif;font-size:14px;font-weight:normal;padding:10px 5px;border-style:solid;border-width:1px;overflow:hidden;word-break:normal;border-color:#ccc;color:#333;background-color:#f0f0f0;}
+		.tg .tg-4eph{background-color:#f9f9f9}
+	</style>
+</head>
```
Comment (VuThiTranVan) [inline src/main/webapp/WEB-INF/views/person.jsp]: Create CSS file, don't write CSS into JSP file

File: src/main/webapp/WEB-INF/views/person.jsp [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+<body>
+<h1>
+	Add a Person
+</h1>
```
Comment (VuThiTranVan) [inline src/main/webapp/WEB-INF/views/person.jsp]: multi-language by creating message.properties file

File: src/main/webapp/index.html [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+          integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
+    <link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.7.0/css/font-awesome.min.css" rel="stylesheet"
+          integrity="sha384-wvfXpqpZZVQGK6TAh5PVlGOfQNHSoD2xbE+QkPxCAFlNEevoEH3Sl0sibVcOQVnN" crossorigin="anonymous">
+	<link href="css/custom.css" rel="stylesheet">
+	<style type="text/css">
+		.ng-hide {
+			display: none;
+		}
+	</style>
+</head>
```
Comment (VuThiTranVan) [inline src/main/webapp/index.html]: Create CSS file, don't write CSS into JSP file

I think you should using Tiles Framework to design template for layout

File: src/main/webapp/partials/create.html [hunk 8]
```diff
@@ -0,0 +1,18 @@
+<div class="page-header">
```
Comment (VuThiTranVan) [inline src/main/webapp/partials/create.html]: using jsp file and using form tag of Spring

File: pom.xml (hunk 1 above)
Comment (tashdidjonayed) [reply pom.xml]: ok I will update

File: src/main/java/com/framgia/fitm/controller/UserController.java [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+		
+		try{
+			user = userService.login(user);
+		}
+		catch (Exception e) {
+			// TODO: handle exception
+		}
+		return user;
+	}
+}
```
[the remark this answers is not in the document]
Comment (tashdidjonayed) [reply src/main/java/com/framgia/fitm/controller/UserController.java]: not complete , only skeleton

File: src/main/java/com/framgia/fitm/dao/UserDAO.java (hunk 2 above)
Comment (tashdidjonayed) [reply src/main/java/com/framgia/fitm/dao/UserDAO.java]: not complete

File: src/main/webapp/WEB-INF/spring/root-context.xml (hunk 4 above)
Comment (tashdidjonayed) [reply src/main/webapp/WEB-INF/spring/root-context.xml]: no not necessary , generated when creating the source

File: src/main/webapp/WEB-INF/views/person.jsp (hunk 5 above)
Comment (tashdidjonayed) [reply src/main/webapp/WEB-INF/views/person.jsp]: noted

**Pull request closed without merging**